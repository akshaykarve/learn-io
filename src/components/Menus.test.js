import React from 'react';
import renderer from 'react-test-renderer';
import MenuController from "./Menus";

const BrowserRouter = require("react-router-dom").BrowserRouter;

it('Not Logged In', () => {
	
	const tree = renderer.create(
	<BrowserRouter>
		<MenuController 
		isSignedIn={false} 
		isAdmin={false} 
		/>
	</BrowserRouter>
	).toJSON();
	expect(tree).toMatchSnapshot();
});
it('Not Logged In But Admin', () => {
	
	const tree = renderer.create(
	<BrowserRouter>
		<MenuController 
		isSignedIn={false} 
		isAdmin={true} 
		/>
	</BrowserRouter>
	).toJSON();
	expect(tree).toMatchSnapshot();
});
it('Logged In', () => {
	
	const tree = renderer.create(
	<BrowserRouter>
		<MenuController 
		isSignedIn={true} 
		isAdmin={false} 
		/>
	</BrowserRouter>
	).toJSON();
	expect(tree).toMatchSnapshot();
});
it('Logged In And Admin', () => {
	
	const tree = renderer.create(
	<BrowserRouter>
		<MenuController 
		isSignedIn={true} 
		isAdmin={true} 
		/>
	</BrowserRouter>
	).toJSON();;
	expect(tree).toMatchSnapshot();
});


export default MenuController;