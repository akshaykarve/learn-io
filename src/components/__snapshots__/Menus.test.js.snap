// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logged In 1`] = `
<div
  className="menusStyle"
>
  <a
    className="homeButton"
    href="/home"
    onClick={[Function]}
  >
     Learn-io 
  </a>
  <div
    className="dropdown"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropbtn dropdown-toggle btn btn-primary"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="menus"
        height="40px"
        src="menus.png"
        width="40px"
      />
    </button>
  </div>
</div>
`;

exports[`Logged In And Admin 1`] = `
<div
  className="menusStyle"
>
  <a
    className="homeButton"
    href="/home"
    onClick={[Function]}
  >
     Learn-io 
  </a>
  <div
    className="dropdown"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropbtn dropdown-toggle btn btn-primary"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="menus"
        height="40px"
        src="menus.png"
        width="40px"
      />
    </button>
  </div>
</div>
`;

exports[`Not Logged In 1`] = `
<div
  className="menusStyle"
>
  <a
    className="homeButton"
    href="/home"
    onClick={[Function]}
  >
     Learn-io 
  </a>
  <div
    className="dropdown"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropbtn dropdown-toggle btn btn-primary"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="menus"
        height="40px"
        src="menus.png"
        width="40px"
      />
    </button>
  </div>
</div>
`;

exports[`Not Logged In But Admin 1`] = `
<div
  className="menusStyle"
>
  <a
    className="homeButton"
    href="/home"
    onClick={[Function]}
  >
     Learn-io 
  </a>
  <div
    className="dropdown"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropbtn dropdown-toggle btn btn-primary"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="menus"
        height="40px"
        src="menus.png"
        width="40px"
      />
    </button>
  </div>
</div>
`;
